compiler/rustc_span/src/hygiene.rs

```diff
@@ -1188,6 +1188,53 @@ impl MacroKind {
     }
 }
 
+/// A set of macro kinds, for macros that can have more than one kind
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Encodable, Decodable, Hash, Debug)]
+#[derive(HashStable_Generic)]
+pub struct MacroKinds(u8);
```

Comment:
The structure isn't actually used in `rustc_span`.
Maybe move it to some higher crate in the dependency tree?

Reply:
@petrochenkov It seems like it would be a little confusing for `MacroKind` and `MacroKinds` to be in two different crates.

That said, the four crates that use `MacroKinds` are `rustc_expand`, `rustc_hir`, `rustc_metadata`, and `rustc_resolve`.
- `rustc_expand` depends on `rustc_hir`
- `rustc_metadata` depends on `rustc_hir` and `rustc_expand`
- `rustc_resolve` depends on `rustc_hir` and `rustc_expand` and `rustc_metadata`

So, of those, the only one that might make sense is `rustc_hir`. I can move it there.